fix: Extend character limit for API, query, and datasource names

saiprabhu-dandanayak opened the pull request:
## Description of the PR

### Bug : [Bug Link](https://github.com/appsmithorg/appsmith/issues/6489)
 I have raised this PR inorder to `To extends the character limit for API, query, and datasource names, allowing for more descriptive and clearer identifiers in the system.`

### 📸 Screenshoots
#### here we gave query name more than 30 characters
![image](https://github.com/zemoso-int/appsmith-from-the-business/assets/122865888/147f2ab1-7c90-4cab-b8a5-6cd24853ce2e)

#### When you hover over a query that isn't visible, a tooltip appears displaying the query's name as below.
![querynamesimage](https://github.com/zemoso-int/appsmith-from-the-business/assets/122865888/417f6191-b1d0-4e16-987a-28374dc10392)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Input fields across various components now support a maximum length of 256 characters and utilize full width for better user experience.

- **Enhancements**
  - Improved name validation logic for entity names to simplify and enhance performance.
  - Added special character removal for certain input fields to ensure cleaner data entries.

- **Bug Fixes**
  - Fixed potential issues with callback dependencies to ensure more accurate updates and functionality.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ayushpahwa commented on app/client/src/components/editorComponents/ActionNameEditor.tsx:
```diff
@@ -1,4 +1,4 @@
-import React, { memo } from "react";
+import React, {memo} from "react";
```
Is this a prettier miss?

saiprabhu-dandanayak replied:
fixed.

ayushpahwa reviewed (changes requested):
No description for changes or a connected issue and the prettier seems to be failing.

saiprabhu-dandanayak commented:
> No description for changes or a connected issue and the prettier seems to be failing.

Hi @ayushpahwa , added description along with the bug link.

saiprabhu-dandanayak commented:
> > No description for changes or a connected issue and the prettier seems to be failing.
> 
> Hi @ayushpahwa , added description along with the bug link.

Hi @ayushpahwa , I've just pulled the latest changes from the release branch , fixed prettier linting issue in the current one , raised an updated pull request (PR) to reflect these changes , please review if you have enough bandwidth , please let me know your feedback.

ayushpahwa commented on app/client/src/components/utils/NameEditorComponent.tsx:
```diff
@@ -127,7 +124,7 @@ function NameEditor(props: NameEditorProps) {
       }
     },
-    [dispatch, isInvalidNameForEntity, entityId, entityName],
+    [hasActionNameConflict, entityName, suffixErrorMessage],
```
Whats the reason to remove the dependencies here?

ayushpahwa commented on app/client/src/utils/helpers.tsx:
```diff
@@ -172,13 +172,9 @@ function removeClass(ele: HTMLElement, cls: string) {
-    .slice(0, limit || 30);
-};
+export const removeSpecialChars = (value: string) => {
+  return value.replace(/\W+/g, '_'); 
```
Why remove the separator regex here?

saiprabhu-dandanayak replied:
i haven't removed it , it was suggested by coderabbitai  to write in a more readable way , the functionality is being the same , but removed the slicing upto 30 characters.

ayushpahwa reviewed (changes requested):
Thanks for the fixes. Apologies for the delay, I was OOO. I've re-review the PR. Left 2 comments. Also, this might fail tests, I'm triggering the CI. @rohan-arthur can you also help if this is going to affect any ongoing projects? Thanks!

saiprabhu-dandanayak commented:
> Thanks for the fixes. Apologies for the delay, I was OOO. I've re-review the PR. Left 2 comments. On top of that, please add some unit tests and this might fail some existing tests (I'm triggering the CI).
> 
> @rohan-arthur can you also help if this is going to affect any ongoing projects? Thanks!

ok @ayushpahwa , will add unit test for the current changes done.

hetunandu commented:
We have not made any decision on issue and no assignment was made for the same. A character limit of 256 is too high and needs more thought. Closing this PR
